src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -2112,13 +2112,13 @@ private bool SkipComment()
                 }
                 else
                 {
-                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.UnexpectedEndOfDataWithinStartOfComment);
+                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.UnexpectedEndOfDataWhileReadingComment);
```

Comment:
~Use `InvalidCharacterAtStartOfComment` here. `UnexpectedEndOfDataWhileReadingComment` doesn't make sense for this case.~

OR, better yet: since this is "Skip" and not "Consume", keep `ExpectedStartOfValueNotFound` here.

Reply:
Skip in Multi-segment does the right thing. Single-segment should do the same:
https://github.com/dotnet/corefx/blob/372fb7e9bcbf297b0de8e4c91e6f208e512c7939/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs#L2179
https://github.com/dotnet/corefx/blob/372fb7e9bcbf297b0de8e4c91e6f208e512c7939/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs#L2186
https://github.com/dotnet/corefx/blob/372fb7e9bcbf297b0de8e4c91e6f208e512c7939/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs#L2206

Reply:
The value sent is `JsonConstants.Slash` but that could be an `*` as well, which we don't know yet because it is not there. Essentially, it could be `/ or *` and we don't have placeholders for two characters in that exception. This is where all my thought process to have all those separate exception messages started.

So was the original exception that just emitted a `/` correct?

Reply:
At line 2179 and 2186, the buffer length is already zero. We don't know the intention yet whether it was to follow with a `/` or `*` for single / multi line comment respectively. That is why I considered `ExpectedStartOfComment` as a wrong message. Because it expects a `/` (only). Why not `*`? We can't say anything if there is nothing!

At 2206, I did get confused because the exception resource name says `ExpectedStartOfValueNotFound` where as the actual message is `<value> is invalid start of a value.`. Perhaps, the exception resource name should be `UnexpectedStartOfValue` to align with the contents of exception message.

Reply:
I see the confusion. Maybe I can try to help explain my perspective.

The reason why value sent is `JsonConstants.Slash` is because before calling Skip(), we already saw a slash (that's what got us into the Skip method). See:
https://github.com/dotnet/corefx/blob/6f9570e012054bf407778c0b07632e7733ecbb13/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs#L895-L897

It doesn't matter if `*` follows next or `/`. Skip mode tries to ignore comments all together. So, the exception user sees is, we saw `/` and then either end of data or something invalid. `/` is not a valid start of value which is what we would expect (think of skip as if comments don't exist at all). That's why we have the `ExpectedStartOfValueNotFound` message.

> That is why I considered `ExpectedStartOfComment` as a wrong message. 

For "Skip" mode, I agree, we shouldn't return `ExpectedStartOfComment`. I am suggesting we return `ExpectedStartOfValueNotFound`. I don't think it makes sense for the user to see a comment related message when the user said "skip" them.

Let's say you have invalid JSON like this: `{"property": /}`. I expect an exception saying `/` is invalid start of a value. We should throw that regardless of whether the user was in "skip" or "disallow".

> So was the original exception that just emitted a `/` correct?

In my opinion, yes.

> Perhaps, the exception resource name should be `UnexpectedStartOfValue` to align with the contents of exception message.

Maybe, but I don't feel strongly about it. If you think it makes it clearer, I would be ok with renaming it.

Reply:
@ahsonkhan I get the idea now. Thank you for the clarification. Didn't realize the subtlety in two usages of one exception message.

I have submitted the changes. Hope I have finally got them as wanted. Thanks for patience.